src/arrayMethodSort.js

```diff
@@ -4,8 +4,31 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const defaultCompareFun = (a, b) => {
+      const strA = String(a);
+      const strB = String(b);
+
+      return strA < strB ? -1 : strA > strB ? 1 : 0;
+    };
+
+    const compare =
+      typeof compareFunction === 'function'
+        ? compareFunction
+        : defaultCompareFun;
+
+    for (let i = 0; i < this.length - 1; i++) {
+      for (let j = 0; j < this.length - 1 - i; j++) {
```

Comment:
The outer loop should iterate until this.length, not this.length - 1. In the current implementation, the largest element will not be correctly sorted if it starts at the beginning of the array.